src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,56 @@
+package duke.command;
+
+import duke.error.DukeException;
+import duke.error.ErrorTypes;
+import duke.error.ThrowError;
+import duke.output.Printer;
+import duke.task.Task;
+
+import java.util.ArrayList;
+
+public class Command {
+    public static void evaluate(String input, ArrayList<Task> tasks) {
+        try {
+            String[] arrayOfInput = input.split(" ");
+            if (input.equals(CommandWords.LIST.COMMAND)) {
+                Printer.listTasks(tasks);
+            } else if (CheckMarkUnmark.isMark(arrayOfInput)) {
+                // if command is "mark <int>"
+                CommandActions.markTask(tasks, arrayOfInput);
+            } else if (CheckMarkUnmark.isUnmark(arrayOfInput)) {
+                // if command is "unmark <int>"
+                CommandActions.unmarkTask(tasks, arrayOfInput);
+            } else if (CheckDelete.isDelete(arrayOfInput)) {
+                // if command is "delete <int>"
+                CommandActions.deleteTask(tasks, arrayOfInput);
+            } else {
+                // command is to add task
+                decideTaskType(input, tasks, arrayOfInput);
+            }
+        } catch (DukeException e) {
+            Printer.endLine();
+        }
+    }
+
+    public static void decideTaskType(String input, ArrayList<Task> tasks, String[] arrayOfInput) throws DukeException {
```

Comment:
decide task type is also doing validity check. ideally these 2 are 2 separate operations. 
please consider.